python/gtsam/examples/SFMExample_bal.py

```diff
@@ -7,112 +7,122 @@
   See LICENSE for the license information
 
   Solve a structure-from-motion problem from a "Bundle Adjustment in the Large" file
-  Author: Frank Dellaert (Python: Akshay Krishnan, John Lambert)
+  Author: Frank Dellaert (Python: Akshay Krishnan, John Lambert, Varun Agrawal)
 """
 
 import argparse
 import logging
 import sys
 
-import matplotlib.pyplot as plt
-import numpy as np
-
 import gtsam
-from gtsam import (
-    GeneralSFMFactorCal3Bundler,
-    PinholeCameraCal3Bundler,
-    PriorFactorPinholeCameraCal3Bundler,
-    readBal,
-    symbol_shorthand
-)
+from gtsam import (GeneralSFMFactorCal3Bundler,
+                   PriorFactorPinholeCameraCal3Bundler, PriorFactorPoint3)
+from gtsam.symbol_shorthand import C, P
+from gtsam.utils import plot
+from matplotlib import pyplot as plt
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+DEFAULT_BAL_DATASET = "dubrovnik-3-7-pre"
+
+
+def plot_scene(scene_data: gtsam.SfmData, result: gtsam.Values):
```

Comment:
looks good. nit here:  missing return type

Reply:
Sorry, I don't see the utility of adding a `None` return type specification. Unless there is a compelling reason for this, we are losing the forest for the trees.

Reply:
For me, it makes it easier to immediately understand the function signature.

If we don't have it, the type annotations are not complete, and we can't rely on mypy or other static analyzers completely.

Reply:
If it's missing, we should be able to assume default None if the docstring doesn't mention a return type. No return type == None makes sense.

Reply:
We can infer that, but mypy can't : - )

Reply:
Mypy doesn't report None violations regardless. I added `-> None` to the run function and did `x = run(parser.parse_args())`, then I ran mypy and mypy didn't report any issue here, which is the correct behavior.

This goes back to my original comment, this would be useful for autocomplete. However, we're missing the main point that the wrapper doesn't provide type definitions out of the box, so mypy will fail on a lot of other things related to GTSAM before a missing return type None. Plus, we don't even check for type definitions in the CI, so again, I don't see a compelling reason for this review request.

Reply:
I added in the type-hints.